django/db/backends/base/schema.py

```diff
@@ -942,6 +942,9 @@ def _constraint_names(self, model, column_names=None, unique=None,
         Returns all constraint names matching the columns and conditions
         """
         column_names = list(column_names) if column_names else None
+        if column_names and self.connection.features.uppercases_column_names:
+            # Oracle's get_constraints() returns all columns as lower case.
```

Comment:
It turns out that there is an [existing function](https://github.com/django/django/blob/6d61ec0e1a4eb5768be5add9e1c44c89dacbfa7e/django/db/backends/base/introspection.py#L29-L42) to transform column names when comparing to values returned by introspection. I'm now using this instead which should be correct without examining the features flag.